include/openssl/asn1t.h

```diff
@@ -661,21 +671,21 @@ typedef int ASN1_ex_d2i(ASN1_VALUE **pval, const unsigned char **in, long len,
                         const ASN1_ITEM *it, int tag, int aclass, char opt,
                         ASN1_TLC *ctx);
 
-typedef int ASN1_ex_i2d(ASN1_VALUE **pval, unsigned char **out,
+typedef int ASN1_ex_i2d(const ASN1_VALUE **pval, unsigned char **out,
                         const ASN1_ITEM *it, int tag, int aclass);
 typedef int ASN1_ex_new_func(ASN1_VALUE **pval, const ASN1_ITEM *it);
 typedef void ASN1_ex_free_func(ASN1_VALUE **pval, const ASN1_ITEM *it);
 
-typedef int ASN1_ex_print_func(BIO *out, ASN1_VALUE **pval,
+typedef int ASN1_ex_print_func(BIO *out, const ASN1_VALUE **pval,
```

Comment:
API change.

Reply:
Agreed, the double `**` ist problematic. Would `const ASN1_VALUE * const *pval` be acceptable?

Reply:
> Agreed, the double `**` ist problematic.

_(Note: I am thinking of http://c-faq.com/ansi/constmismatch.html)_

Reply:
> Would `const ASN1_VALUE * const *pval` be acceptable?

Not sure anymore, this might only work in C++.

Reply:
I wanted to point to that ...

Reply:
> I wanted to point to that ...

Oh, thanks nevertheless. I overlooked that the C++ case was mentioned explicitly in the above c-faq link.